lib/statsd.js

```diff
@@ -464,5 +462,37 @@ Client.prototype.childClient = function(options) {
   return new ChildClient(this, options);
 };
 
+// Overrides tags in parent with tags from child with the same name (case sensitive) and return the result as new
+// array. parent and child are not mutated.
+function overrideTags (parent, child) {
+  var childCopy = {};
+  var toAppend = [];
+  child.forEach(function (tag) {
+    var idx = tag.indexOf(':');
```

Comment:
Well, if you mean `tag` could be `null` or `undefined` then IMO it should blow up because it is a [programming error](https://www.joyent.com/developers/node/design/errors). That way user can fix the bug in his/her code and it is better than adding meaningless tags. WDYT?